Added support for front panel port prefix regex

itamar-talmon opened the pull request:
- What I did
Removed the dependency on the "Ethernet" string in the SONiC code base and added support
for extending the front panel port name pattern.

- How I did it
1. Introduced FRONT_PANEL_PORT_PREFIX_REGEX that extends the old FRONT_PANEL_PORT_PREFIX ("Ethernet")
2. Updated all the relevant usage of the "Ethernet" throughout the code base to use the new regex pattern

- How to verify it
Pass all UT and CI testing.

- Why I did it
In order to support distinguishing between different types of front panel ports in a maintainable fashion.
Specifically, we are planning to bring up a system with 'service' ports (in addition to the regular ethernet data ports) - these
are lower speed ports that used for connection to accelerators, internal loopbacks and more.

**- Related Commits and Merge Strategy**
_This is part of a group of related commits and should be merged **after https://github.com/Azure/sonic-swss-common/pull/598**._

The full merge order is:

1. swss-common - https://github.com/Azure/sonic-swss-common/pull/598
2. sonic-buildimage - https://github.com/Azure/sonic-buildimage/pull/10471
3. swsssdk - https://github.com/Azure/sonic-py-swsssdk/pull/121
4. all the rest
     [ https://github.com/Azure/sonic-utilities/pull/2127](https://github.com/Azure/sonic-utilities/pull/2127)
     [ https://github.com/Azure/sonic-snmpagent/pull/251](https://github.com/Azure/sonic-snmpagent/pull/251)
     [ https://github.com/Azure/sonic-swss/pull/2223](https://github.com/Azure/sonic-swss/pull/2223)
     [ https://github.com/Azure/sonic-platform-daemons/pull/252](https://github.com/Azure/sonic-platform-daemons/pull/252)
     [ https://github.com/Azure/sonic-platform-common/pull/274](https://github.com/Azure/sonic-platform-common/pull/274)

itamar-talmon commented:
This PR depends on https://github.com/Azure/sonic-swss-common/pull/598 so it will not pass until it merged

qiluo-msft commented on rules/sonic-py-common.mk:
```diff
@@ -4,7 +4,8 @@ ifeq ($(ENABLE_PY2_MODULES), y)
     $(SONIC_PY_COMMON_PY2)_SRC_PATH = $(SRC_PATH)/sonic-py-common
     $(SONIC_PY_COMMON_PY2)_DEPENDS += $(SWSSSDK_PY2)
-    $(SONIC_PY_COMMON_PY2)_PYTHON_VERSION = 2
+    $(SONIC_PY_COMMON_PY2)_DEBS_DEPENDS += $(PYTHON_SWSSCOMMON)
```
>PYTHON_SWSSCOMMON [](http://example.com/codeflow?start=45&length=17)

Thanks for fixing the dependency! It is not related to front panel port prefix. Could you raise a separate PR for the fix? We may need to backport the fix.

itamar-talmon replied:
The reason I have to add the dependency here is because I needed the new port panel regex from swsscomon in sonic_py_common, if you think it's worth to make it a separate PR I can do it.

qiluo-msft replied:
The dependency fix itself is a good bug fix. Please help.

itamar-talmon replied:
Sure, no problem, i'll create a new PR with only the dependency fix and send it to you.

itamar-talmon replied:
Actually this was fixed in master few weeks ago by - https://github.com/Azure/sonic-buildimage/commit/e4ed475fcde0d68abb3b2814a71b76296642a7ee.
I just took the latest master and removed my change.

qiluo-msft commented on src/sonic-config-engine/portconfig.py:
```diff
@@ -258,6 +259,14 @@ def __init__(self, name, bmode, properties):
 
         if not self._breakout_capabilities:
             raise RuntimeError("Unsupported breakout mode {}!".format(bmode))
+    def _parse_name_to_prefix_and_id(self):
```
>_parse_name_to_prefix_and_id [](http://example.com/codeflow?start=8&length=28)

Leave 1~2 lines before and after this function. Not use 0 or too many empty lines.

itamar-talmon replied:
done

qiluo-msft commented on src/sonic-config-engine/portconfig.py:
```diff
@@ -258,6 +259,14 @@ def __init__(self, name, bmode, properties):
         if not self._breakout_capabilities:
             raise RuntimeError("Unsupported breakout mode {}!".format(bmode))
+    def _parse_name_to_prefix_and_id(self):
+        match = re.search(swsscommon.FRONT_PANEL_PORT_PREFIX_REGEX + "(\d+)", self._name)
```
>FRONT_PANEL_PORT_PREFIX_REGEX [](http://example.com/codeflow?start=37&length=29)

There is an implicit assumption: there must be one and only one group in FRONT_PANEL_PORT_PREFIX_REGEX. This is not flexible and not easy to change in future.

Better to capture the prefix and id into one regex, no concatenating.

itamar-talmon replied:
done

itamar-talmon commented:
@kcudnik @qiluo-msft - could you please check it out after my update?
Checks will fail until related commits will be merged (in the description).
Thank you!

qiluo-msft commented on src/sonic-config-engine/portconfig.py:
```diff
@@ -259,6 +260,13 @@ def __init__(self, name, bmode, properties):
+        match = re.match(swsscommon.FRONT_PANEL_PORT_REGEX, self._name)
+        try:
+            return match.group(1), int(match.group(2))
+        except Exception:
```
>Exception [](http://example.com/codeflow?start=15&length=9)

Please use more specific exception type.

itamar-talmon replied:
I removed the exception due to the above comment

qiluo-msft commented on src/sonic-config-engine/portconfig.py:
```diff
@@ -259,6 +260,13 @@ def __init__(self, name, bmode, properties):
 
+    def _parse_name_to_prefix_and_id(self):
+        match = re.match(swsscommon.FRONT_PANEL_PORT_REGEX, self._name)
+        try:
```
>try [](http://example.com/codeflow?start=8&length=3)

Just check `match` instead of `try/except`?

itamar-talmon replied:
Yes, good idea, i'll do that

itamar-talmon commented:
@kcudnik @qiluo-msft - I updated the code with the new comments and rebased since the dependency fix was already merged to master by a different commit (you can check my comment there). Please check, thank you!
